packages/@datapull/redshift-destination/index.js

```diff
@@ -0,0 +1,85 @@
+const processConfig = require('@datapull/json-config').processConfig;
+const Client = require('pg').Client;
+
+class RedshiftDestination {
+  constructor(config) {
+    // replace variables in the config:
+    try {
+      this.config = processConfig(config);
+    } catch (err) {
+      throw Error(`Could not parse config of Redshift Destination ${err}`);
+    }
+
+    ['user', 'host', 'database', 'password', 'port'].forEach(param => {
+      if (!this.config[param]) {
+        throw Error(`[Redshift destination] Parameter "${param}" is required for this destination`);
+      }
+    });
+
+    this.clientConfig = {
+      user: this.config.user,
+      host: this.config.host,
+      database: this.config.database,
+      password: this.config.password,
+      ssl: true,
+      port: this.config.port
+    };
+  }
+
+  buildPipelines(pipelines) {
+    pipelines.forEach(p => {
+      p.destination = {
+        name: ['Redshift', this.config.host, this.config.table].join(':'),
+        runner: this.insertNewRows.bind(this)
+      };
+    });
+
+    return pipelines;
+  }
+
+  async insertNewRows(messages) {
+    const client = new Client(this.clientConfig);
+
+    await client.connect();
+
+    let insertCount = 0;
+    let skipCount = 0;
+
+    for (let i in messages) {
```

Comment:
I would prefer using `forEach` unless we somehow use `i` later on.

Reply:
`for` loop will do all iterations sequentially and will proceed to next line only when all async operations inside are done. That's a different behaviour from `forEach`, which would require additional code using `Promise.all` to wait for all the asynchronous calls being done

Reply:
And `for .. of` is not supported, am I right?

Reply:
it is supported, i will change the code to use it